qiskit/algorithms/state_fidelities/compute_uncompute.py

```diff
@@ -0,0 +1,118 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Compute-uncompute fidelity interface using primitives
+"""
+
+from __future__ import annotations
+from collections.abc import Sequence
+from copy import copy
+
+from qiskit import QuantumCircuit
+from qiskit.primitives import BaseSampler
+from .base_state_fidelity import BaseStateFidelity
+from .state_fidelity_result import StateFidelityResult
+
+
+class ComputeUncompute(BaseStateFidelity):
+    """
+    This class leverages the sampler primitive to calculate the fidelity of
+    two quantum circuits with the compute-uncompute method.
+    """
+
+    def __init__(self, sampler: BaseSampler, **run_options) -> None:
+        r"""
+        Initializes the class to evaluate the compute-uncompute state fidelity using
+        the sampler primitive. This fidelity is defined as the state overlap
+
+            :math:`|\langle\psi(x)|\phi(y)\rangle|^2`,
+
+        where :math:`x` and :math:`y` are optional parametrizations of the
+        states :math:`\psi` and :math:`\phi` prepared by the circuits
+        ``circuit_1`` and ``circuit_2``, respectively.
+        Args:
+            sampler: Sampler primitive instance.
+            run_options: Backend runtime options used for circuit execution.
```

Comment:
Can we elaborate on this being the default set etc. The gradients have more info (I'll copy it here for reference)

```
            run_options: Backend runtime options used for circuit execution. The order of priority is:
                run_options in ``run`` method > gradient's default run_options > primitive's default
                setting. Higher priority setting overrides lower priority setting.
```